### PR TITLE
add delegate userDidSkipThisVersion

### PR DESCRIPTION
**Pull request by BobZombie**
I added it because I wanted a delegate when the user pressed the  "skip This Version" button. 
I am happy if you did merge it.
Thank you.

### CODE REVIEW

**Comment by kornelski** [comment]
Seems like a reasonable thing to add.  But please check if the method is implemented before calling it. See how other delegate methods do it.

File: Sparkle/SUUIBasedUpdateDriver.m [hunk 1]
```diff
@@ -162,6 +163,7 @@ - (void)updateAlertFinishedWithChoice:(SUUpdateAlertChoice)choice forItem:(SUApp
             break;
 
         case SUSkipThisVersionChoice:
+            [[updater delegate] userDidSkipThisVersion:self.updater];
```
Comment (Neoster) [inline Sparkle/SUUIBasedUpdateDriver.m]: One should check for responsiveness before calling the method on the delegate.

`if ([[updater delegate] respondsToSelector:@selector(userDidSkipThisVersion:)]) {
        [[updater delegate] updater:self.updater userDidSkipThisVersion:self.updateItem];
    }`

**Comment by Neoster** [comment]
@BobZombie are you going to process the requested changes? Need any help with that?

**Comment by kornelski** [comment]
Thanks

File: Sparkle/SUUIBasedUpdateDriver.m [hunk 2]
```diff
@@ -162,6 +163,9 @@ - (void)updateAlertFinishedWithChoice:(SUUpdateAlertChoice)choice forItem:(SUApp
             break;
 
         case SUSkipThisVersionChoice:
+            if ([[updater delegate] respondsToSelector:@selector(userDidSkipThisVersion:)]) { 
+                [[updater delegate] updater:self.updater userDidSkipThisVersion:self.updateItem];
```
Comment (LeoNatan) [inline Sparkle/SUUIBasedUpdateDriver.m]: How did this pass review? respondsToSelector is checked with one selector, while another is called inside. The inside selector is unknown, causing compile error.

https://github.com/sparkle-project/Sparkle/issues/1455

**Pull request merged**